mergify_engine/context.py

```diff
@@ -203,58 +207,68 @@ def get_config_location_cache_key(
     ) -> str:
         return f"config-location~{owner_login}~{repo_name}"
 
-    async def get_mergify_config_content(
+    async def iter_mergify_config_files(
         self,
-        ref: typing.Optional[github_types.GitHubRefType] = None,
-    ) -> typing.Union[typing.Tuple[None, None], typing.Tuple[str, bytes]]:
+        ref: typing.Optional[github_types.SHAType] = None,
+        preferred_filename: typing.Optional[str] = None,
+    ) -> typing.AsyncIterator[MergifyConfigFile]:
         """Get the Mergify configuration file content.
 
         :return: The filename and its content.
         """
 
-        if ref is None and "mergify_config" in self._cache:
-            return self._cache["mergify_config"]
-
-        config_location_cache = self.get_config_location_cache_key(
-            self.installation.owner_login, self.name
-        )
-
         kwargs = {}
         if ref:
             kwargs["ref"] = ref
-            cached_filename = None
-        else:
-            cached_filename = await self.installation.redis.get(config_location_cache)
 
         filenames = self.MERGIFY_CONFIG_FILENAMES.copy()
-        if cached_filename:
-            filenames.remove(cached_filename)
-            filenames.insert(0, cached_filename)
+        if preferred_filename:
+            filenames.remove(preferred_filename)
+            filenames.insert(0, preferred_filename)
 
         for filename in filenames:
             try:
-                content = (
+                content = typing.cast(
+                    github_types.GitHubContentFile,
                     await self.installation.client.item(
                         f"{self.base_url}/contents/{filename}",
                         **kwargs,
-                    )
-                )["content"]
+                    ),
+                )
             except http.HTTPNotFound:
                 continue
-            if ref is None and filename != cached_filename:
+
+            yield MergifyConfigFile(
+                type=content["type"],
+                content=content["content"],
+                path=content["path"],
+                sha=content["sha"],
+                decoded_content=base64.b64decode(
+                    bytearray(content["content"], "utf-8")
+                ),
+            )
```

Comment:
I guess you could cast the dict + add the new key?
Not a big deal, that just creating a new dict is fine too :)